pkg/db/vulnerability_detail.go

```diff
@@ -13,11 +13,24 @@ const (
 	vulnerabilityDetailBucket = "vulnerability-detail"
 )
 
+func GetVulnerabilityType(cveID string) (string, error) {
+	for _, prefix := range types.VulnerabilityTypes {
+		if strings.HasPrefix(strings.ToUpper(cveID), prefix) {
+			return prefix, nil
+		}
+	}
+	return "Unknown", nil
+}
+
 func (dbc Config) PutVulnerabilityDetail(tx *bolt.Tx, cveID, source string, vuln types.VulnerabilityDetail) error {
 	root, err := tx.CreateBucketIfNotExists([]byte(vulnerabilityDetailBucket))
 	if err != nil {
 		return err
 	}
+	vuln.IdType, err = GetVulnerabilityType(cveID)
+	if err != nil {
+		return err
+	}
```

Comment:
I don't think we should change the data in `PutVulnerabilityDetail` since it is just a thin wrapper of BoltDB.
I think it is better to put `IdType` in `VulnerabilityDetail` of each `VulnSrc` .

```
vuln := types.VulnerabilityDetail{
	Description: oval.Description,
	IdType: types.GetVulnerabilityType(vulnID,)
	References:  referencesFromContains(references, []string{elsaID, vulnID}),
	Title:       oval.Title,
	Severity:    severityFromThreat(oval.Severity),
}
```
This code makes it possible to verify the value of IdType as an argument of  PutVulnerabilityDetail in unit tests.